src/matcha_ml/state/remote_state_manager.py

```diff
@@ -142,6 +142,9 @@ def is_state_provisioned(self) -> bool:
         if not self._configuration_file_exists():
             return False
 
+        if not self.azure_storage.resource_group_exists:
```

Comment:
For consistency, can we create a function `self._resource_group_exists` similar to `_bucket_exists`?

Reply:
A test when `is_state_provisioned` returns False when resource group is not present?